java-checks/src/main/java/org/sonar/java/checks/ReplaceLambdaByMethodRefCheck.java

```diff
@@ -87,16 +83,32 @@ private static boolean isBlockWithOneStatement(Tree tree) {
     return tree.is(Tree.Kind.BLOCK) && ((BlockTree) tree).body().size() == 1;
   }
 
-  private static boolean isExpressionStatementInvokingMethod(Tree statement) {
-    return statement.is(Tree.Kind.EXPRESSION_STATEMENT) && isMethodInvocation(((ExpressionStatementTree) statement).expression());
+  private static boolean isExpressionStatementInvokingMethod(Tree statement, LambdaExpressionTree lambdaTree) {
+    return statement.is(Tree.Kind.EXPRESSION_STATEMENT) && isMethodInvocation(((ExpressionStatementTree) statement).expression(), lambdaTree);
   }
 
-  private static boolean isReturnStatementInvokingMethod(Tree statement) {
-    return statement.is(Tree.Kind.RETURN_STATEMENT) && isMethodInvocation(((ReturnStatementTree) statement).expression());
+  private static boolean isReturnStatementInvokingMethod(Tree statement, LambdaExpressionTree lambdaTree) {
+    return statement.is(Tree.Kind.RETURN_STATEMENT) && isMethodInvocation(((ReturnStatementTree) statement).expression(), lambdaTree);
   }
 
-  private static boolean isMethodInvocation(@Nullable Tree tree) {
-    return tree != null && tree.is(Tree.Kind.METHOD_INVOCATION);
+  private static boolean isMethodInvocation(@Nullable Tree tree, LambdaExpressionTree lambdaTree) {
+    if (tree != null && tree.is(Tree.Kind.METHOD_INVOCATION, Tree.Kind.NEW_CLASS)) {
+      Arguments arguments;
+      if(tree.is(Tree.Kind.NEW_CLASS)) {
+        if(((NewClassTree) tree).classBody() != null) {
+          return false;
+        }
+        arguments = ((NewClassTree) tree).arguments();
+      } else {
+        arguments = ((MethodInvocationTree) tree).arguments();
+      }
+      return arguments.size() == lambdaTree.parameters().size() &&
+        IntStream.range(0, arguments.size()).allMatch(i -> {
+          List<IdentifierTree> usages = lambdaTree.parameters().get(i).symbol().usages();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 9. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)